frameworks/GAMA/exec.py

```diff
@@ -0,0 +1,82 @@
+import logging
+import os
+import sys
+import tempfile as tmp
+
+if sys.platform == 'darwin':
+    os.environ['OBJC_DISABLE_INITIALIZE_FORK_SAFETY'] = 'YES'
+os.environ['JOBLIB_TEMP_FOLDER'] = tmp.gettempdir()
+os.environ['OMP_NUM_THREADS'] = '1'
+os.environ['OPENBLAS_NUM_THREADS'] = '1'
+os.environ['MKL_NUM_THREADS'] = '1'
+from gama import GamaClassifier, GamaRegressor, __version__
+import sklearn
+import category_encoders
+
+from frameworks.shared.callee import call_run, result, Timer
+
+
+log = logging.getLogger(__name__)
+
+
+def run(dataset, config):
+    log.info("\n**** GAMA  %s ****", __version__)
+    log.info("sklearn == %s", sklearn.__version__)
+    log.info("category_encoders == %s", category_encoders.__version__)
+
+    is_classification = (config.type == 'classification')
+    # Mapping of benchmark metrics to GAMA metrics
+    metrics_mapping = dict(
+        acc='accuracy',
+        auc='roc_auc',
+        f1='f1',
+        logloss='neg_log_loss',
+        mae='neg_mean_absolute_error',
+        mse='neg_mean_squared_error',
+        msle='neg_mean_squared_log_error',
+        r2='r2'
+    )
+    scoring_metric = metrics_mapping[config.metric] if config.metric in metrics_mapping else None
+    if scoring_metric is None:
+        raise ValueError("Performance metric {} not supported.".format(config.metric))
+
+    training_params = {k: v for k, v in config.framework_params.items() if not k.startswith('_')}
+    n_jobs = config.framework_params.get('_n_jobs', config.cores)  # useful to disable multicore, regardless of the dataset config
+
+    dir, file = os.path.split(config.output_predictions_file)
+    file_name, file_ext = os.path.splitext(file)
+    log_file = os.path.join(dir, file_name + '.gamalog')
```

Comment:
This pattern is used multiple times in integration code each time we need to generate predictions in different format, for logs or whatever.
Wondering if I should add a property like `config.output_file_pattern` or sth similar:
```log_file = config.output_file_pattern.format(ext="gamalog")```

Note that the output dir is already directly available as `config.output_dir` otherwise.

Reply:
I missed the `config.output_dir` and will use that instead.